File: app/src/main/java/ru/org/adons/cblock/ui/view/PermFragment.java
package ru.org.adons.cblock.ui.view;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.provider.Settings;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import ru.org.adons.cblock.R;
import ru.org.adons.cblock.ui.base.BaseFragment;
import ru.org.adons.cblock.ui.base.IBaseFragmentListener;

/**
 * Permission View, show warning if required permissions were not granted
 */
public class PermFragment extends BaseFragment<IBaseFragmentListener> {

    static final String PERM_FRAGMENT_TAG = "PERM_FRAGMENT_TAG";

    @BindView(R.id.button_settings) Button buttonSetting;
    private Unbinder unbinder;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        setListener(activity, IBaseFragmentListener.class);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_perm, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        buttonSetting.setOnClickListener(v -> showSetting());
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }

    private void showSetting() {
        final Intent intent = new Intent();
        intent.setAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
        intent.addCategory(Intent.CATEGORY_DEFAULT);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
        intent.setData(Uri.parse("package:" + getActivity().getPackageName()));
        startActivity(intent);
    }

}

File: app/src/main/java/ru/org/adons/cblock/ui/view/perm/PermFragment.kt
package ru.org.adons.cblock.ui.view.perm

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.provider.Settings
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_perm.*
import ru.org.adons.cblock.R
import ru.org.adons.cblock.ui.activity.BaseFragment
import ru.org.adons.cblock.ui.activity.IMainListener


/**
 * Permission View, show warning if required permissions were not granted
 */
class PermFragment : BaseFragment<IMainListener>() {

    companion object {
        val PERM_FRAGMENT_TAG = "PERM_FRAGMENT_TAG"
    }

    @Suppress("OverridingDeprecatedMember", "DEPRECATION")
    override fun onAttach(activity: Activity?) {
        super.onAttach(activity)
        activity?.let { setListener(activity, IMainListener::class.java) }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? = inflater
            .inflate(R.layout.fragment_perm, container, false)

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        buttonSettings.setOnClickListener { showSetting() }
    }

    private fun showSetting() {
        val intent = Intent()
        intent.action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
        intent.addCategory(Intent.CATEGORY_DEFAULT)
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
        intent.data = Uri.parse("package:" + activity.packageName)
        startActivity(intent)
    }

}
